lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -20,131 +20,157 @@ module Requests
 
-      sig { override.params(node: SyntaxTree::DefNode).void }
-      def visit_def(node)
-        if @modifier == "public"
-          method_name = node.name.value
-          if method_name.start_with?("test_")
-            add_code_lens(
-              node,
-              name: method_name,
-              command: BASE_COMMAND + @path + " --name " + method_name,
-            )
+        sig { params(node: SyntaxTree::Command).void }
+        def on_command(node)
+          if ACCESS_MODIFIERS.include?(node.message.value) && node.arguments.parts.any?
+            @prev_visibility = @visibility
+            @visibility = node.message.value
           end
         end
-      end
 
-      sig { override.params(node: SyntaxTree::Command).void }
-      def visit_command(node)
-        if node.message.value == "public"
-          with_visiblity("public", node)
+        sig { params(node: SyntaxTree::Command).void }
+        def after_command(node)
+          @visibility = @prev_visibility
         end
-      end
 
-      sig { override.params(node: SyntaxTree::CallNode).void }
-      def visit_call(node)
-        ident = node.message if node.message.is_a?(SyntaxTree::Ident)
+        sig { params(node: SyntaxTree::CallNode).void }
+        def on_call(node)
+          ident = node.message if node.message.is_a?(SyntaxTree::Ident)
 
-        if ident
-          if T.cast(ident, SyntaxTree::Ident).value == "public"
-            with_visiblity("public", node)
+          if ident
+            ident_value = T.cast(ident, SyntaxTree::Ident).value
+            if ACCESS_MODIFIERS.include?(ident_value)
+              @prev_visibility = @visibility
+              @visibility = ident_value
+            end
           end
         end
-      end
 
-      sig { override.params(node: SyntaxTree::VCall).void }
-      def visit_vcall(node)
-        vcall_value = node.value.value
+        sig { params(node: SyntaxTree::CallNode).void }
+        def after_call(node)
+          @visibility = @prev_visibility
+        end
+
+        sig { params(node: SyntaxTree::VCall).void }
+        def on_vcall(node)
+          vcall_value = node.value.value
 
-        if ACCESS_MODIFIERS.include?(vcall_value)
-          @modifier = vcall_value
+          if ACCESS_MODIFIERS.include?(vcall_value)
+            @prev_visibility = vcall_value
+            @visibility = vcall_value
+          end
         end
       end
 
-      private
-
-      sig do
-        params(
-          visibility: String,
-          node: T.any(SyntaxTree::CallNode, SyntaxTree::Command),
-        ).void
+      sig { params(other: Listener[ResponseType]).returns(T.self_type) }
+      def merge_response!(other)
+        @response.concat(other.response)
+        self
       end
-      def with_visiblity(visibility, node)
-        current_visibility = @modifier
-        @modifier = visibility
-        visit(node.arguments)
-      ensure
-        @modifier = T.must(current_visibility)
+
+      class << self
```

Comment:
I think some people prefer static methods for things which don't depends on state, but we can move it:

https://github.com/Shopify/ruby-lsp/pull/680

Reply:
Oh, I see. Anything that would be useful for extensions or other requests should be placed in `Requests::Common`, so that it's accessible to everybody.

Reply:
I'll open a PR to move it to there.